### PR TITLE
feat: resolve templates in env section

### PR DESCRIPTION
**Pull request by m4ver1k**
This commit will support resolving templates in Env section

https://github.com/goreleaser/goreleaser/issues/1820

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -522,6 +524,45 @@ type Project struct {
 	GiteaURLs GiteaURLs `yaml:"gitea_urls,omitempty"`
 }
 
+func (p *Project) executeEnvTemplates() error {
```
Comment (caarlos0) [inline pkg/config/config.go:527]: we have our own template package (`tmpl`) with some defaults et al, please use it instead.

example usages can be found in the codebase: https://github.com/goreleaser/goreleaser/search?q=%22tmpl.New%28%22

Comment (caarlos0) [reply pkg/config/config.go:527]: also, the template should be evaluated in the build pipe, not here.

**Comment by caarlos0** [comment]
Thanks for the PR, made a couple of comments above.

File: pkg/config/config.go (hunk 1 above)
Comment (m4ver1k) [reply pkg/config/config.go:527]: Sure will make these changes,
thanks

**Comment by caarlos0** [comment]
superseeded by #2246

thanks for the pr though! :)

**Pull request closed without merging**